### PR TITLE
CI: run and upload Coverage on Coveralls every sunday at midnight

### PR DESCRIPTION
**Pull request by khancyr**

### CODE REVIEW

**Comment by khancyr** [comment]
need https://github.com/ArduPilot/ardupilot/pull/16557

**Comment by amilcarlucas** [comment]
Should this also be added to the DevCall ?

**Comment by khancyr** [comment]
@peterbarker what about merging this ? I won't lock the CI for long time (around 2h on sunday) and would allow use to start using coverage. That is more informative than anything else for now. There isn't any implication on PR

File: Tools/scripts/run-coverage.sh [hunk 1]
```diff
@@ -20,6 +21,21 @@ OPTS="--speedup=$SPEEDUP --timeout=$TIMEOUT --debug --no-clean"
...
+    python -m pip --disable-pip-version-check show "$1" | grep -c "Version"
+}
+
+if [ $(package_is_installed "mavproxy") -eq 0 ]; then
+    python -m pip install --user mavproxy
+    python -m pip uninstall -y pymavlink
+fi
+
+
+if [ $(package_is_installed "pymavlink") -eq 0 ]; then
```
Comment (peterbarker) [inline Tools/scripts/run-coverage.sh:34]: We've seen this elsewhere and asked for it to be fixed.

**Review by peterbarker — CHANGES_REQUESTED** [review]

**Comment by peterbarker** [comment]
I should also point out that if the output doesn't include coverage of all of the rangefinder backends then it is absolutely flawed.  That's why I stopped updating the stuff on the firmware server.

File: Tools/scripts/run-coverage.sh (hunk 1 above)
Comment (khancyr) [reply Tools/scripts/run-coverage.sh]: fixed

**Comment by khancyr** [comment]
let me merge that and I will fix the whole coverage things ! ;-P

File: Tools/scripts/run-coverage.sh [hunk 2]
```diff
@@ -20,6 +21,21 @@ OPTS="--speedup=$SPEEDUP --timeout=$TIMEOUT --debug --no-clean"
...
+    python -m pip install --user mavproxy
+    python -m pip uninstall -y pymavlink
+fi
+
+
+if ! package_is_installed "pymavlink"; then
+    git submodule update --init --recursive
+    (cd modules/mavlink/pymavlink && MDEF="$PWD/../message_definitions" python -m pip install --user .)
+fi
+
```
Comment (peterbarker) [inline Tools/scripts/run-coverage.sh:38]: Hang about.  I run this script on my boxes and I don't want it stuffing around with my environment.

Why would `run_coverage.sh` need to do this?!  Whatever sets up the environment it runs in should be responsible.

**Comment by khancyr** [comment]
closing, it will be integrated into another pr

**Pull request closed without merging**